presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotBrokerPageSource.java

```diff
@@ -0,0 +1,383 @@
+    }
+
+    private static Double parseDouble(String value)
+    {
+        try {
+            return Double.valueOf(value);
+        }
+        catch (NumberFormatException ne) {
+            switch (value) {
+                case PINOT_INFINITY:
+                case PINOT_POSITIVE_INFINITY:
+                    return PRESTO_INFINITY;
+                case PINOT_NEGATIVE_INFINITY:
+                    return PRESTO_NEGATIVE_INFINITY;
+            }
+            throw new PinotException(PinotErrorCode.PINOT_DECODE_ERROR, Optional.empty(), "Cannot decode double value from pinot " + value, ne);
+        }
+    }
+
+    private void setValue(Type type, BlockBuilder blockBuilder, String value)
+    {
+        if (value == null) {
+            blockBuilder.appendNull();
+            return;
+        }
+        if (!(type instanceof FixedWidthType) && !(type instanceof VarcharType)) {
+            throw new PinotException(PinotErrorCode.PINOT_UNSUPPORTED_COLUMN_TYPE, Optional.empty(), "type '" + type + "' not supported");
+        }
+        if (type instanceof FixedWidthType) {
+            completedBytes += ((FixedWidthType) type).getFixedSize();
+            if (type instanceof BigintType) {
+                type.writeLong(blockBuilder, parseDouble(value).longValue());
+            }
+            else if (type instanceof IntegerType) {
+                blockBuilder.writeInt(parseDouble(value).intValue());
+            }
+            else if (type instanceof TinyintType) {
+                blockBuilder.writeByte(parseDouble(value).byteValue());
+            }
+            else if (type instanceof SmallintType) {
+                blockBuilder.writeShort(parseDouble(value).shortValue());
+            }
+            else if (type instanceof BooleanType) {
+                type.writeBoolean(blockBuilder, Boolean.valueOf(value));
+            }
+            else if (type instanceof DecimalType || type instanceof DoubleType) {
+                type.writeDouble(blockBuilder, parseDouble(value));
+            }
+            else if (type instanceof TimestampType) {
+                type.writeLong(blockBuilder, Long.valueOf(value));
+            }
+            else {
+                throw new PinotException(PinotErrorCode.PINOT_UNSUPPORTED_COLUMN_TYPE, Optional.empty(), "type '" + type + "' not supported");
+            }
+        }
+        else {
+            Slice slice = Slices.utf8Slice(value);
+            blockBuilder.writeBytes(slice, 0, slice.length()).closeEntry();
+            completedBytes += slice.length();
+        }
+    }
+
+    private void setValuesForGroupby(List<BlockBuilder> blockBuilders, List<Type> types, int numGroupByClause, JsonNode group, String[] values)
+    {
+        for (int k = 0; k < group.size(); k++) {
+            setValue(types.get(k), blockBuilders.get(k), group.get(k).asText());
+        }
+        for (int aggrIndex = 0; aggrIndex < values.length; ++aggrIndex) {
```

Comment:
just call it `index` and use `index++`. (We don't use `++i` for whatever culture reason 👯 )